src/mono/mono/mini/interp/interp.c

```diff
@@ -7433,32 +7478,38 @@ interp_run_finally_with_il_state (gpointer il_state_ptr, int clause_index, gpoin
 	if (header->num_locals)
 		memset (frame_locals (&frame) + imethod->local_offsets [0], 0, imethod->locals_size);
 	/* Copy locals from il_state */
-	int locals_start = sig->hasthis + sig->param_count;
+	int locals_start = findex;
 	for (int i = 0; i < header->num_locals; ++i) {
 		if (il_state->data [locals_start + i])
 			stackval_from_data (header->locals [i], (stackval*)(frame_locals (&frame) + imethod->local_offsets [i]), il_state->data [locals_start + i], FALSE);
 	}
 
 	memset (&clause_args, 0, sizeof (FrameClauseArgs));
 	clause_args.start_with_ip = (const guint16*)handler_ip;
-	clause_args.end_at_ip = (const guint16*)handler_ip_end;
-	clause_args.exit_clause = clause_index;
+	if (clause_type == MONO_EXCEPTION_CLAUSE_NONE || clause_type == MONO_EXCEPTION_CLAUSE_FILTER)
+		clause_args.end_at_ip = (const guint16*)clause_args.start_with_ip + 0xffffff;
```

Comment:
Is this saying if we're in a filter or a catch clause to run until the end of the method?

Reply:
Yes.

Reply:
For filter clauses, it will run until ENDFILTER.